src/Controller/ViewerController.php

```diff
@@ -63,63 +63,42 @@ public function __construct(
    *   TRUE to open Collabora Online in edit mode.
    *   FALSE to open Collabora Online in readonly mode.
    *
+   * @throws \Symfony\Component\HttpKernel\Exception\BadRequestHttpException
+   *   The Collabora server or editor is not available, or mismatching scheme.
+   *
    * @return \Symfony\Component\HttpFoundation\Response
    *   Response suitable for iframe, without the usual page decorations.
    */
-  public function editor(MediaInterface $media, Request $request, $edit = FALSE) {
+  public function editor(MediaInterface $media, Request $request, $edit = FALSE): Response {
     try {
       $wopi_client_url = $this->discovery->getWopiClientURL();
     }
     catch (CollaboraNotAvailableException $e) {
-      $this->logger->warning(
-        "Collabora Online is not available.<br>\n" . Error::DEFAULT_ERROR_MESSAGE,
-        Error::decodeException($e) + [],
-      );
-      return new Response(
-        (string) $this->t('The Collabora Online editor/viewer is not available.'),
-        Response::HTTP_BAD_REQUEST,
-        ['content-type' => 'text/plain'],
-      );
+      $this->logger->warning('Collabora Online is not available.');
+      throw new BadRequestHttpException('The Collabora Online editor/viewer is not available.');
     }
 
     $current_request_scheme = $request->getScheme();
-    if (!str_starts_with($wopi_client_url, $current_request_scheme . '://')) {
-      $this->logger->error($this->t(
+    if (parse_url($wopi_client_url, PHP_URL_SCHEME) !== $current_request_scheme) {
+      $this->logger->error(
```

Comment:
Ok for me to use `parse_url()` here.

Good catch on removing the `$this->t()` call! I don't know why I missed it.